samples/bluetooth/hci_usb/ doesn't build for nrf5340dk_nrf5340_cpuapp

**Describe the bug**
The sample no longer builds without error on `nrf5340dk_nrf5340_cpuapp`. 

**To Reproduce**
Steps to reproduce the behavior:
1. west build -b nrf5340dk_nrf5340_cpuapp samples/bluetooth/hci_usb/
2. See error

Build error:
```
/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/bluetooth/libdrivers__bluetooth.a(rpmsg.c.obj): in function `bt_rpmsg_close':
/zephyr/drivers/bluetooth/hci/rpmsg.c:342: undefined reference to `bt_hci_cmd_send_sync'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Expected behavior**
Expect the sample to work with `nrf5340dk_nrf5340_cpuapp` as it has before

**Impact**
Annoyance

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: 39433f0669f1fd620f36605d1c3df61a6cf97ae2

**Additional context**
N/A

--- a/drivers/bluetooth/hci/rpmsg.c
+++ b/drivers/bluetooth/hci/rpmsg.c
@@ -339,10 +339,12 @@ static int bt_rpmsg_close(void)
 {
 	int err;
 
-	err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, NULL);
-	if (err) {
-		LOG_ERR("Sending reset command failed with: %d", err);
-		return err;
+	if (IS_ENABLED(CONFIG_BT_HCI_HOST)) {
+		err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, NULL);
+		if (err) {
+			LOG_ERR("Sending reset command failed with: %d", err);
+			return err;
+		}
 	}
 
 	err = ipc_service_deregister_endpoint(&hci_ept);

